internal/utils/configs.go

```diff
@@ -0,0 +1,101 @@
+package utils
+
+import (
+	"context"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	"github.com/covid19cz/erouska-backend/internal/secrets"
+	"github.com/sethvargo/go-envconfig"
+	urlutils "net/url"
+)
+
+//KeyServerConfig Configuration of KeyServer.
+type KeyServerConfig struct {
+	URL string `env:"KEY_SERVER_URL, required"`
+}
+
+//VerificationServerConfig Configuration of Verification server.
+type VerificationServerConfig struct {
+	AdminURL  string `env:"VERIFICATION_SERVER_ADMIN_URL, required"`
+	DeviceURL string `env:"VERIFICATION_SERVER_DEVICE_URL, required"`
+	AdminKey  string
+	DeviceKey string
+}
+
+//LoadKeyServerConfig Load KeyServer config.
+func LoadKeyServerConfig(ctx context.Context) (*KeyServerConfig, error) {
+	logger := logging.FromContext(ctx)
+
+	var keyServerConfig KeyServerConfig
+	if err := envconfig.Process(ctx, &keyServerConfig); err != nil {
+		logger.Debugf("Could not load KeyServerConfig: %v", err)
```

Comment:
Debugf -> Errorf?

Reply:
Well... The idea behind is that I pass the error to the caller and it's up to him what do do with it - probably call `logger.Errorf`. I think it's good practice to not log errors that are passed further on too high level.

Reply:
Eh? I'm talking about error level logging in our backend, I'm not talking about any error handling on callers/clients. Everything in our backend has debug level logging, including errors. That does not make any sense to me. We should properly log - debug, info, warning, error...

Reply:
You got me wrong. On this level, this is logged as debug. However, the error is propagated to the upper level to **caller of this function** (not speaking about a client as device) and it could be logged as error there.